internal/worker.go

```diff
@@ -361,7 +362,10 @@ func replayWorkflowHistory(logger *zap.Logger, service workflowserviceclient.Int
 		Logger:   logger,
 	}
 	taskHandler := newWorkflowTaskHandler(domain, params, nil, getHostEnvironment())
-	_, _, err := taskHandler.ProcessWorkflowTask(&workflowTask{task: task, historyIterator: iterator})
+	resp, _, err := taskHandler.ProcessWorkflowTask(&workflowTask{task: task, historyIterator: iterator})
```

Comment:
Seems to me that we should throw nondeterministic error inside ProcessWorkflowTask.
Can you check if https://github.com/uber-go/cadence-client/pull/709 solves this issue?